test/android/README.md

```diff
@@ -1,96 +0,0 @@
-# On-Target Unit Testing [Android]
```

Comment:
We need to add a section in `how-to-build.md` on how to run the tests on Android.